db/kvstore_shards.go

```diff
@@ -0,0 +1,111 @@
+package db
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/db/batch"
+)
+
+type (
+	// KVStoreShards is the interface of sharded KV store
+	KVStoreShards interface {
+		KVStore
+		Shard(bucketPrefix string) KVStore
+	}
+	kvStoreShards struct {
+		KVStore
+		once sync.Once
+	}
+	kvStorePrefix struct {
+		kv           KVStore
+		bucketPrefix string
+	}
+)
+
+// NewKVStoreShards creates a new KVStoreSharding
+func NewKVStoreShards(kv KVStore) KVStoreShards {
+	return &kvStoreShards{
+		KVStore: kv,
+	}
+}
+
+func (k *kvStoreShards) Start(ctx context.Context) error {
+	k.once.Do(func() {
+		if err := k.KVStore.Start(ctx); err != nil {
+			panic(fmt.Sprintf("failed to start underlying kvstore: %v", err))
+		}
+	})
+	return nil
+}
+
+func (k *kvStoreShards) Stop(ctx context.Context) error {
+	k.once.Do(func() {
+		if err := k.KVStore.Stop(ctx); err != nil {
```

Comment:
`once` would only invoke `Do()` one time for the lifetime of the `once` instance, so the `Stop()` here won't be called

Reply:
need 2 `once`, one for Start and the other for Stop